nav2_amcl/include/nav2_amcl/amcl_node.hpp

```diff
@@ -206,7 +207,8 @@ class AmclNode : public nav2_util::LifecycleNode
    * @brief Initialize state services
    */
   void initServices();
-  rclcpp::Service<std_srvs::srv::Empty>::SharedPtr global_loc_srv_;
+  std::shared_ptr<nav2_util::ServiceServer<std_srvs::srv::Empty,
```

Comment:
We could make this nicer if the `ServiceServer` had a typedef for `SharedPtr` so this could be changed to `nav2_util::ServiceServer<std_srvs::srv::Empty, std::shared_ptr<nav2_util::LifecycleNode>>>::SharedPtr`

That could then be used everywhere the type is declared